### ISSUE TITLE
Replacement bet status doesn't update when error is 'BET_TAKEN_OR_LAPSED'

### ISSUE THREAD

**Issue by new-to-python99** [betcode-org/flumine#455]
In live, when a bet is replaced, but the original bet had already been replaced this can result in a 'BET_TAKEN_OR_LAPSED' error, however when this error occurs, the order status remains as OrderStatus.REPLACING and never updates, which means no further bets are possible on this runner until the process is killed and re-started.

It appears to be a very specific issue that may only occur with Best Price Execution turned off (or possibly when a bet is replaced and then attempted be replaced again within quick succession).

I will try to create a dummy strategy to replicate the issue, but as of yet have been unable to.

**Comment by flxbe** [comment]
Hi! I noticed a related issue when cancelling an order that has already been matched. But I think the underlying issue is the same. I created a PR to solve the issue for the cancel case, but I think a similar fix should be possible for replacing orders. So far I am waiting for feedback from Liam, but as he said in slack, he is currently taking some time off.

I think a strategy to replicate the issue could be to:
1. Get the `bet_id` from an already matched order
2. Insert the order as `executable` into the blotter (a little hacky)
3. Try to replace/cancel the order

Since I am encountering this issue quite often I would be happy to assist in any way to resolve it. Right now, I am using a hacky workaround by overwriting the `BetfairExecution` class.

**Comment by new-to-python99** [comment]
Good plan, that makes sense as effectively a replace is simply cancelling and then placing a new order (albeit with some linkage under the hood), so it makes sense that your scenario would have the same issue!

**Comment by new-to-python99** [comment]
Still having problems with this (as discussed recently in the Slack Group), however it appears that the issue arises randomly on when cancelling an order, not just when replacing one. However I still can't spot why it occurs sometimes and not other times.

Here is a brief log extract, where we can see the order fails to cancel (which makes sense as it has already been cancelled), but the bit that doesn't make sense is why sometimes it gets stuck in this perpetual loop:

`{"asctime": "2022-05-12 18:34:44,479", "levelname": "INFO", "message": "Trade status update: Pending", "id": "1e516c34-d222-11ec-bc32-2016d8944b85", "strategy": "STRATEGY_NAME", "place_reset_seconds": 0.0, "reset_seconds": 0.0, "orders": ["138716356370423222"], "offset_orders": [], "notes": "", "market_notes": null, "status": "Pending", "status_log": "Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending"}

INFO:flumine.execution.baseexecution:Order Cancel: FAILURE

{"asctime": "2022-05-12 18:34:44,504", "levelname": "INFO", "message": "Order Cancel: FAILURE", "bet_id": "268025195271", "order_id": "138716356370423222", "status": "FAILURE", "error_code": "BET_TAKEN_OR_LAPSED"}

INFO:flumine.order.order:Order status update: Executable

{"asctime": "2022-05-12 18:34:44,509", "levelname": "INFO", "message": "Order status update: Executable", "market_id": "1.186353923", "selection_id": 36210, "handicap": 0, "id": "138716356370423222", "customer_order_ref": "3d1ca481c20a8-138716356370423222", "bet_id": "268025195271", "date_time_created": "2022-05-12 08:07:18", "publish_time": null, "market_version": null, "async": null, "trade": {"id": "1e516c34-d222-11ec-bc32-2016d8944b85", "strategy": "STRATEGY_NAME", "place_reset_seconds": 0.0, "reset_seconds": 0.0, "orders": ["138716356370423222"], "offset_orders": [], "notes": "", "market_notes": null, "status": "Pending", "status_log": "Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending, Live, Pending"}, "order_type": {"order_type": "Limit", "price": 8.6, "size": 1.98, "persistence_type": "PERSIST", "time_in_force": null, "min_fill_size": null, "bet_target_type": null, "bet_target_size": null}, "info": {"side": "LAY", "size_matched": 0.0, "size_remaining": 1.98, "size_cancelled": 0.0, "size_lapsed": 0.0, "size_voided": 0.0, "average_price_matched": 0.0}, "responses": {"date_time_placed": "2022-05-12 08:07:18", "elapsed_seconds_executable": null}, "runner_status": null, "status": "Executable", "status_log": "Pending, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable, Cancelling, Executable", "violation_msg": null, "simulated": {"profit": 0.0, "piq": 0.0, "matched": []}, "notes": "", "market_notes": null, "client": "CLIENT"}`

**Comment by liampauling** [comment]
Hmm so flumine thinks the order has some `size_remaining` hence the loop, either there is a race condition somewhere or the order stream has failed, does it coincide with a downtime?

How long is the order live before you cancel?

**Comment by new-to-python99** [comment]
It varies, I'm not 100% if there us a pattern with regards to cancel time, but I did notice one at least had a creation time > 2 hours before attempting to cancel (that may sound odd, but for some of the illiquid politics markets the price can remain unchanged for days)!

I don't believe there's any correlation to downtime from what I see, but I have never seen it on 'quick markets' e.g. horse racing, only longer ones (e.g. politics or tournament outright winner types).

I did think it may be when an order is added to the blotter via middleware (i.e. it exists from a previous flumine session), but I tested that and it still occurs if the order was created during that flumine session.

Sorry I appreciate that's not much help!!
